TKGame/UI/GameMenu.cs

```diff
@@ -1,15 +1,183 @@
+                image.Id = name;
+
+			return image;
+		}
+
+		private void AddWidgetToHeaderPanel(Widget widget, string text = null, bool isVisible = true)
+        {
+            Grid newGrid = new Grid();
+
+            newGrid.ColumnsProportions.Add(new Proportion(ProportionType.Auto));
+            newGrid.RowsProportions.Add(
+                new Proportion() 
+                {
+                    Type = ProportionType.Pixels,
+                    Value = 60
+                });
+            newGrid.RowsProportions.Add(new Proportion(ProportionType.Auto));
+            newGrid.RowSpacing = 10;
+            newGrid.ColumnSpacing = 10;
+            newGrid.VerticalAlignment = VerticalAlignment.Bottom;
+            newGrid.Border = new SolidBrush(Color.White);
+            newGrid.BorderThickness = new Myra.Graphics2D.Thickness(0, 0, 2, 2);
+            newGrid.Padding = new Myra.Graphics2D.Thickness(5);
+			newGrid.Background = new SolidBrush(new Color(Color.Black, 50));
+            newGrid.Visible = isVisible;
+
+
+			widget.GridColumn = 0;
+            widget.GridRow = 0;
+            widget.HorizontalAlignment = HorizontalAlignment.Center;
+            widget.VerticalAlignment = VerticalAlignment.Bottom;
+            newGrid.Widgets.Add(widget);
+
+            if (text is not null)
+            {
+                Label widgetLabel = new Label();
+                widgetLabel.Text = text;
+                widgetLabel.Font = fontSystem.GetFont(fontSize);
+                widgetLabel.HorizontalAlignment = HorizontalAlignment.Center;
+                widgetLabel.GridColumn = 0;
+                widgetLabel.GridRow = 1;
+
+                newGrid.Widgets.Add(widgetLabel);
+            }
+
+			panelHsp.Widgets.Add(newGrid);
+        }
+
+        public void Update()
+        {
+            int playerHp = Player.Instance.health;
+
+			playerHealthLabel.Text = playerHp.ToString();
+            playerHealthLabel.TextColor = (playerHp > 0) ? Color.LimeGreen : Color.Red;
+
+            if (!playerGotPowerup && Player.Instance.GetAttackComponents().Count > 0)
+            {
+                playerGotPowerup = true;
+                EnableAttackComponentWidgets();
+            }
+        }
+
+        private void EnableAttackComponentWidgets()
+        {
+            foreach (Widget widget in panelHsp.Widgets.Where(w => !w.Visible))
+                widget.Visible = true;
+        }
+
+        public void ChangeWeaponTexture(Texture2D texture)
+        {
+            foreach (Widget widget in panelHsp.Widgets.Where(w => w is Grid))
+            {
+
+                if ((widget as Grid).Widgets.Any(w => w is Image && w.Id == weaponImageId))
+                {
+                    Image image = (widget as Grid).Widgets.First(w => w is Image && w.Id == weaponImageId) as Image;
+                    image.Renderable = new TextureRegion(texture, new Rectangle(0, 0, texture.Width, texture.Height));
+                }
+            }
+        }
```

Comment:
I can tell this is similar to the other menu classes inheriting from IMenu, though making a habit of adding comments (even more than necessary) is something we all need to do for readability.